bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -0,0 +1,345 @@
+        return UnDefType.UNDEF;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // we can not update single channels and refresh is triggered automatically
+            // anyways
+            return;
+        }
+
+        String key = null;
+        String value = null;
+        switch (channelUID.getId()) {
+            case MAX_AMPERE:
+                key = "amp";
+                if (command instanceof DecimalType) {
+                    value = String.valueOf(((DecimalType) command).intValue());
+                } else if (command instanceof QuantityType<?>) {
+                    value = String.valueOf(
+                            ((QuantityType<ElectricCurrent>) command).toUnit(SmartHomeUnits.AMPERE).intValue());
+                }
+                break;
+            case SESSION_CHARGE_CONSUMPTION_LIMIT:
+                key = "dwo";
+                if (command instanceof DecimalType) {
+                    value = String.valueOf(((DecimalType) command).intValue() * 10);
+                } else if (command instanceof QuantityType<?>) {
+                    value = String.valueOf(
+                            ((QuantityType<Energy>) command).toUnit(SmartHomeUnits.KILOWATT_HOUR).intValue() * 10);
+                }
+                break;
+            case ALLOW_CHARGING:
+                key = "alw";
+                if (command instanceof OnOffType) {
+                    value = command == OnOffType.ON ? "1" : "0";
+                }
+                break;
+            default:
+        }
+        if (key != null && value != null) {
+            sendData(key, value);
+        } else {
+            logger.warn("Could not update channel {} with key {} and value {}", channelUID.getId(), key, value);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(GoEChargerConfiguration.class);
+        allChannels = getThing().getChannels().stream().map(channel -> channel.getUID().getId())
+                .collect(Collectors.toList());
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        startAutomaticRefresh();
+        logger.debug("Finished initializing!");
+    }
+
+    private String getUrl(String type) {
+        String urlStr = type;
+        return urlStr.replace("%IP%", StringUtils.trimToEmpty(config.ip));
+    }
+
+    private void sendData(String key, String value) {
+        String urlStr = getUrl(GoEChargerBindingConstants.MQTT_URL).replace("%KEY%", key).replace("%VALUE%", value);
+        logger.debug("POST URL = {}", urlStr);
+
+        String result = null;
+        try {
+            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(HttpMethod.POST)
+                    .timeout(5, TimeUnit.SECONDS).send();
+
+            result = contentResponse.getContentAsString();
+            // TODO check if value really changed
```

Comment:
I think you can remove this comment. If you did not receive an error, you can assume that it worked.  And the next refresh will show the updated state.
```suggestion
```

Reply:
Guess you are right here...